toml/src/main/kotlin/org/rust/toml/annotator/CargoTomlErrorAnnotator.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.annotator
+
+import com.intellij.ide.annotator.AnnotatorBase
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.psi.PsiElement
+import org.rust.toml.CargoTomlDependencyVisitor
+import org.rust.toml.crates.getCrate
+import org.toml.lang.psi.TomlKey
+import org.toml.lang.psi.TomlValue
+
+class CargoTomlErrorAnnotator : AnnotatorBase() {
+    override fun annotateInternal(element: PsiElement, holder: AnnotationHolder) {
+        object : CargoTomlDependencyVisitor() {
+            override fun visitDependency(name: TomlKey, version: TomlValue?) {
+                checkDependency(name, version, holder)
+            }
+        }.visitElement(element)
+    }
+
+    private fun checkDependency(dependency: TomlKey, version: TomlValue?, holder: AnnotationHolder) {
+        val crate = getCrate(dependency)
+
+        if (crate == null) {
+            holder.annotateError(dependency, "Crate doesn't exist")
+            return
+        }
+
+        // Version must be present past this point (it could've been not present because the user didn't type it yet)
+        if (version == null) {
+            return
+        }
+
+        // TODO: use version req
+        val versionValue = version.text.removeSurrounding("\"")
+        val versionInfo = crate.versions.find { it.version.rawVersion == versionValue }
```

Comment:
Unfortunately, it produces false positive
For example, patch number of dependency version can be omitted.
![image](https://user-images.githubusercontent.com/2539310/91031795-bee51600-e609-11ea-8d30-e5233b2e5cb9.png)

Moreover, you can add some requirements (see https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements) to a dependency version instead of exact version.
![image](https://user-images.githubusercontent.com/2539310/91032031-11263700-e60a-11ea-89c5-763c121f87a8.png)
![image](https://user-images.githubusercontent.com/2539310/91032091-23a07080-e60a-11ea-8df6-acb259918a5d.png)